From db4ab955e444a9e45d155094be60e97b702cfd09 Mon Sep 17 00:00:00 2001
From: Zhihao Cheng <chengzhihao1@huawei.com>
Date: Wed, 16 Nov 2022 10:53:02 +0800
Subject: [PATCH 0041/1398] [Huawei] Revert "[Backport] file: fix kabi change
 since add f_ctl_mode"

Offering: HULK
hulk inclusion
category: feature
bugzilla: 187975

--------------------------------

This reverts commit cd75e353871049119c9ea6cf9ee8acbe49e607d5.

Signed-off-by: Zhihao Cheng <chengzhihao1@huawei.com>
---
 include/linux/fs.h | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/include/linux/fs.h b/include/linux/fs.h
index 5c71b826c89c..63d451380042 100644
--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -954,6 +954,7 @@ struct file {
 	atomic_long_t		f_count;
 	unsigned int 		f_flags;
 	fmode_t			f_mode;
+	fmode_t			f_ctl_mode;
 	struct mutex		f_pos_lock;
 	loff_t			f_pos;
 	struct fown_struct	f_owner;
@@ -975,14 +976,8 @@ struct file {
 	struct address_space	*f_mapping;
 	errseq_t		f_wb_err;
 	errseq_t		f_sb_err; /* for syncfs */
-#ifndef __GENKSYMS__
-	union {
-		fmode_t			f_ctl_mode;
-		u64			kabi_reserved1;
-	};
-#else
+
 	KABI_RESERVE(1)
-#endif
 } __randomize_layout
   __attribute__((aligned(4)));	/* lest something weird decides that 2 is OK */
 
-- 
2.17.1

